Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Internal/Payout.hs

```diff
@@ -70,7 +79,28 @@ juspayPayoutWebhookHandler merchantShortId mbOpCity authData value = do
                 QFTB.updatePayoutStatusById (Just $ castPayoutOrderStatus payoutStatus) (Id bookingId)
               fork "Update Payout Status and Transactions for MetroBooking" $ do
                 callPayoutService (Id bookingId) payoutOrderId
-          _ -> throwError $ InternalError $ "Unsupported Payout Entity: " <> show payoutOrder.entityName
+          Just DPayment.REFERRAL_AWARD_RIDE -> do
+            when (payoutStatus `elem` [Payout.SUCCESS, Payout.FULFILLMENTS_SUCCESSFUL]) do
+              personStats <- QPersonStats.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+              QPersonStats.updateReferralAmountPaid (personStats.referralAmountPaid + payoutOrder.amount.amount) personId
+            fork "Update Payout Status and Transactions for Referral Award" $ do
+              callReferralPayoutService payoutOrder payoutConfig
+          Just DPayment.REFERRED_BY_AWARD -> do
+            when (payoutStatus `elem` [Payout.SUCCESS, Payout.FULFILLMENTS_SUCCESSFUL]) do
+              QPersonStats.updateReferredByEarningsPayoutStatus (Just $ castOrderStatus payoutStatus) personId
+            fork "Update Payout Status and Transactions for ReferredBy Award" $ do
+              callReferralPayoutService payoutOrder payoutConfig
+          Just DPayment.BACKLOG -> do
+            when (payoutStatus `elem` [Payout.SUCCESS, Payout.FULFILLMENTS_SUCCESSFUL]) do
+              personStats <- QPersonStats.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+              let mbStatus = Just $ castOrderStatus payoutStatus
+              case payoutOrder.amount.amount of
+                amount_ | amount_ == personStats.backlogPayoutAmount -> QPersonStats.updateBacklogPayoutStatus mbStatus personId
+                amount_ | amount_ == personStats.referredByEarnings -> QPersonStats.updateReferredByEarningsPayoutStatus mbStatus personId
```

Comment:
This can be equal based on what configs we have
personStats.backlogPayoutAmount and personStats.referredByEarnings
So update might not be correct